### PR TITLE
add missing tests

### PR DESCRIPTION
**Pull request by stanislavgalin**

### CODE REVIEW

File: app/test/e2e/layer-get.spec.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    delete createdLayer.status;
+
+    // add fields that are generated on the api side
+    createdLayer.interactionConfig = {};
+    createdLayer.updatedAt = receivedLayer.attributes.updatedAt;
+
+    receivedLayer.attributes.should.deep.equal(createdLayer);
+};
+
+describe('Layers - GET enpoints', () => {
```
Comment (tiagojsag) [inline app/test/e2e/layer-get.spec.js]: typo

File: app/test/e2e/layer-update.spec.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    createMockDataset(datasetId);
+    const layer = createLayer(apps, datasetId, layerId);
+    await new Layer(layer).save();
+
+    return requester
+        .patch(`${layerPrefix}/change-environment/${datasetId}/${env}`)
+        .send({ loggedUser: role });
+};
+
+describe('Layers - PATCH enpoints', () => {
```
Comment (tiagojsag) [inline app/test/e2e/layer-update.spec.js]: typo

File: app/test/e2e/layer-update.spec.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    it('PATCH /dataset/:dataset/layer/:layer - Update specific layer should update', async () => {
+        const datasetLayer = await updateLayer({ role: ROLES.ADMIN }, LAYER_TO_UPDATE);
+        datasetLayer.status.should.equal(200);
+        const responseData = datasetLayer.body.data;
+
+        Object.keys(LAYER_TO_UPDATE)
+            .map(field => responseData.attributes[field].should.deep.equal(LAYER_TO_UPDATE[field]));
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - should return error not found, when dataset is not exist', async () => {
```
Comment (tiagojsag) [inline app/test/e2e/layer-update.spec.js]: can you please spit test files based on endpoint. Having a single file testing multiple endpoints is confusing and hard to debug. thanks

Comment (tiagojsag) [inline app/test/e2e/layer-update.spec.js]: also, `when dataset is not exist` => `when dataset doesn't exist`

File: app/test/e2e/layer-delete.spec.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            .delete(`${datasetPrefix}/123/layer/123?loggedUser=${JSON.stringify(ROLES.ADMIN)}`)
+            .send();
+
+        datasetLayer.status.should.equal(404);
+        ensureCorrectError(datasetLayer.body, 'Layer with id \'123\' doesn\'t exist');
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - should delete the specific layer in specific dataset', async () => {
+        const datasetLayer = await deleteLayer(ROLES.ADMIN, '123');
+        datasetLayer.status.should.equal(200);
```
Comment (tiagojsag) [inline app/test/e2e/layer-delete.spec.js]: please validate the response body content, not just the code. I assume this delete endpoint returns the deleted object in the body, so please ensure it returns the correct object, with the correct structure.

File: app/test/e2e/layer-get.spec.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    it('GET /layer - should return layers, when layers is exist', async () => {
+        const testLayer = createLayer();
+        await new Layer(testLayer).save();
+
+        const list = await requester.get(layerPrefix);
+        list.body.should.have.property('data').and.be.an('array').and.length.above(0);
+        ensureCorrectLayer(list.body.data[0], testLayer);
+    });
+
+    it('GET /layer/:layer - should return error not found, when layer is not exist', async () => {
```
Comment (tiagojsag) [inline app/test/e2e/layer-get.spec.js]: same typo as before

File: app/test/e2e/layer-get.spec.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        const layer = await requester.get(`${layerPrefix}/123`);
+        layer.status.should.equal(404);
+        ensureCorrectError(layer.body, 'Layer with id \'123\' doesn\'t exist');
+    });
+
+    it('GET /layer/:layer - should return layer, when layer is exist', async () => {
```
Comment (tiagojsag) [inline app/test/e2e/layer-get.spec.js]: `when layer is exist` =>`when layer exists`

File: app/test/e2e/layer-update.spec.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        ensureCorrectError(envLayer.body, 'Not authorized');
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - Should update layer env', async () => {
+        const layerId = getUUID();
+
+        const envLayer = await updateEnv({ role: ROLES.MICROSERVICE, layerId });
+        envLayer.status.should.equal(200);
+        const layer = await Layer.findOne({ _id: layerId });
+        expect(layer.env).to.be.equal('test');
```
Comment (tiagojsag) [inline app/test/e2e/layer-update.spec.js]: is there a response body? if so, please test it

Comment (stanislavgalin) [reply app/test/e2e/layer-update.spec.js]: there is an empty body

**Pull request merged**